test/foundry/Certify.t.sol

```diff
@@ -0,0 +1,222 @@
+
+        uint256 certifyUntil = 0;
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Grant CERTIFIER role to Alice
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Expect the Certify event
+        vm.expectRevert(abi.encodeWithSelector(ZeroCertifyUntil.selector, alice));
+
+        // Call the certify function
+        vault.certify(certifyUntil, referenceBlockNumber, forceUntil, data);
+
+        vm.stopPrank();
+    }
+
+    /// Test certify reverts on future reference
+    function testCertifyRevertOnFutureReferenceBlock(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 certifyUntil,
+        bytes memory data,
+        uint256 fuzzedFutureBlockNumber,
+        uint256 blockNumber,
+        bool forceUntil
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        blockNumber = bound(block.number, 0, type(uint256).max);
+        vm.roll(blockNumber);
+        fuzzedFutureBlockNumber = bound(fuzzedFutureBlockNumber, blockNumber + 1, type(uint256).max - 1);
+
+        certifyUntil = bound(certifyUntil, 1, blockNumber);
+
+        // Prank as Alice for the transaction
+        vm.startPrank(alice);
+        OffchainAssetReceiptVault vault = OffchainAssetVaultCreator.createVault(factory, alice, assetName, assetSymbol);
+
+        // Grant CERTIFIER role to Alice
+        vault.grantRole(vault.CERTIFIER(), alice);
+
+        // Expect the Certify event
+        vm.expectRevert(abi.encodeWithSelector(FutureReferenceBlock.selector, alice, fuzzedFutureBlockNumber));
+
+        // Call the certify function
+        vault.certify(certifyUntil, fuzzedFutureBlockNumber, forceUntil, data);
+
+        vm.stopPrank();
+    }
+
+    /// Test certify with force until true
+    function testCertifyWithForceUntilTrue(
+        uint256 fuzzedKeyAlice,
+        string memory assetName,
+        string memory assetSymbol,
+        uint256 certifyUntil,
+        uint256 certifyUntilPast,
+        uint256 referenceBlockNumber,
+        bytes memory data,
+        uint256 blockNumber
+    ) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        blockNumber = bound(block.number, 0, type(uint256).max);
+        vm.roll(blockNumber);
+        referenceBlockNumber = bound(referenceBlockNumber, 0, blockNumber);
+        certifyUntil = bound(certifyUntil, 1, type(uint32).max);
+        vm.assume(certifyUntil > certifyUntilPast && certifyUntilPast != 0);
```

Comment:
is there actually a reason to bound the certify until for the force to be in the past?

we want to know that when force is `true` that we can certify in the past, present and future, not just the past